crates/table/src/layout.rs

```diff
@@ -329,12 +332,36 @@ impl MemoryUsage for PrimitiveType {}
 impl HasLayout for PrimitiveType {
     fn layout(&self) -> &'static Layout {
         match self {
-            Self::Bool | Self::I8 | Self::U8 => &Layout { size: 1, align: 1 },
-            Self::I16 | Self::U16 => &Layout { size: 2, align: 2 },
-            Self::I32 | Self::U32 | Self::F32 => &Layout { size: 4, align: 4 },
-            Self::I64 | Self::U64 | Self::F64 => &Layout { size: 8, align: 8 },
-            Self::I128 | Self::U128 => &Layout { size: 16, align: 16 },
-            Self::I256 | Self::U256 => &Layout { size: 32, align: 32 },
+            Self::Bool | Self::I8 | Self::U8 => &Layout {
+                size: 1,
+                align: 1,
+                fixed: true,
+            },
+            Self::I16 | Self::U16 => &Layout {
+                size: 2,
+                align: 2,
+                fixed: true,
+            },
+            Self::I32 | Self::U32 | Self::F32 => &Layout {
+                size: 4,
+                align: 4,
+                fixed: true,
+            },
+            Self::I64 | Self::U64 | Self::F64 => &Layout {
+                size: 8,
+                align: 8,
+                fixed: true,
+            },
+            Self::I128 | Self::U128 => &Layout {
+                size: 16,
+                align: 16,
+                fixed: true,
+            },
+            Self::I256 | Self::U256 => &Layout {
+                size: 32,
+                align: 32,
+                fixed: true,
+            },
```

Comment:
Consider pulling this into a constructor `Layout::new_primitive(size_and_align: u16) -> Self { Self { size: size_and_align, align: size_and_align, fixed: true } }`?